### ISSUE TITLE
Problems with path length running nox

### ISSUE THREAD

**Issue by sivogel** [hydpy-dev/hydpy#150]
I was having problems running a nox session due to path length issues which were quite difficult to identify. Actually my path to the noxfile was quite short ("D:\PycharmProjects\hydpy_master\noxfile.py"), but running nox leads to a path "D:\PycharmProjects\hydpy_master\. nox\doctest-numpy-1\Lib\site-packages\hydpy\cythons\autogen\_build\Release\PycharmProjects\hydpy_master\.nox\doctest-numpy-1\Lib\site-packages\hydpy\cythons\autogen\callback_sw1d_gate_out_gateheight_adjust_2.cp312-win_amd64.lib" which causes problems.

**Comment by tyralla** [comment]
Yes, I encountered this problem, too.  See 8db87182.  This is a Windows-only problem.  However, I think you can quickly increase the maximum path length on your local computer.  If so, please confirm and close this issue.

**Comment by sivogel** [comment]
Increasing the path length would certainly help (I shortened my path to "D:\PycharmProjects\hydpy" and it worked). But in my opinion the problem is not to solve this problem but to recognise the problem. I got the error message: "LINK : fatal error LNK1104:" which I do not associate with path length problems at first. Might there be a solution to solve this? I also wondered why the part "doctest-numpy-1\Lib\site-packages\hydpy\cythons" appears twice in the path.

**Comment by tyralla** [comment]
The error message mentioned is emitted by the Visual Studio compiler, which is used by Cython.  So, I don't think we can capture the error directly and modify the error message with reasonable effort.   But we could eventually add some logic that checks if the desired pyd file has been generated and prints some hints on general problems that might have caused the build to fail.

The sub-path duplication is due to Cython building temporary working directories.  It would be preferable to shorten these paths, but I did not invest enough time to check if and how this is possible.

However, as we have a simple solution (increase the maximum length), adding a note for Windows users to the documentation is probably also okay.

Then, let's leave this issue open until we agree on (and implement) one of these "solutions".

**Comment by sivogel** [comment]
An other solution might also be to check the path length before running the Visual Studio compiler and give more readable error messages.

**Comment by tyralla** [comment]
If I remember correctly, from HydPy's perspective, the creation of the temporary paths and input files and the subsequent start of the build process are performed in one step (by Cython).
